qiskit/transpiler/passes/utils/contains_instruction.py

```diff
@@ -23,19 +23,23 @@ class ContainsInstruction(AnalysisPass):
     that instruction and ``False`` if it does not.
     """
 
-    def __init__(self, instruction_name):
+    def __init__(self, instruction_name, recurse: bool = True):
```

Comment:
Is there a reason why this is added as a kwarg here but not for `size` or `depth`?

Reply:
Not a necessarily _good_ reason, but: in our current uses, `Depth` and `Size` should always recurse, whereas for things like `ContainInstruction`, if you want to use it to detect control flow, you don't need to pay the cost of recursing the `count_ops` call to get the information you need.

Reply:
Would you prefer to have keyword arguments on the other passes as well?  I can't think of any use-cases for them, though.

Reply:
I think so, if only for consistency with the `DAGCircuit` methods.

Reply:
Done in 74172e00a.